File: Server-CSharp/YT/YT/Connection.cs
using Google.Protobuf;
using System;
using System.Collections.Generic;
using System.Text;
using TKcp;

namespace YT
{
    public class Connection
    {
        //客户端的Peer
        public uint Conv;
        //缓存区
        public ByteArray readBuff = new ByteArray();

        public Server Server;

        public Connection(uint conv) {
            Conv = conv;
        }

        public void Send(IMessage message) {
            Server.Send(Conv, message);
        }
    }
}

File: Server-Go/YT/Net/Connection.go
package Net

import (
	"ReaminHistory/YT/Base"
	"google.golang.org/protobuf/runtime/protoiface"
)

type Connection struct {
	//客户端Peer
	Conv uint32
	//缓存区
	readBuf *Base.ByteArray
	Server  *Server
}

// NewConnection 新建一个Connection
func NewConnection(conv uint32) *Connection {
	connection := new(Connection)
	connection.Conv = conv
	connection.readBuf = new(Base.ByteArray)
	return connection
}

// Send 给这个连接发送消息
func (connection *Connection) Send(message protoiface.MessageV1) {
	connection.Server.Send(connection.Conv, message)
}
